lib/govuk_publishing_components/presenters/shared_helper.rb

```diff
@@ -0,0 +1,23 @@
+module GovukPublishingComponents
+  module Presenters
+    class SharedHelper
+      attr_reader :options, :margin_bottom, :heading_level
+
+      def initialize(local_assigns)
+        @options = local_assigns
+        @margin_bottom = @options[:margin_bottom] || 3
+        @heading_level = @options[:heading_level] || 2
+      end
+
+      def get_margin_bottom
+        [*0..9].include?(@margin_bottom) ? "govuk-!-margin-bottom-#{margin_bottom}" : "govuk-!-margin-bottom-3"
+      end
+
+      def get_heading_level
+        return [*1..6].include?(@heading_level) ? "h#{@heading_level}" : "h2" unless @heading_level.zero?
+
+        "span"
```

Comment:
The logic basically says return `h1` to `h6` if passed a number between 1 and 6, otherwise return `h2`, unless zero is passed. At which point the code continues, and the last thing the method returns is a span. The image card component has an option where you can either have a heading or pass a zero for no heading at all, which is why this is in. 

An alternative way of writing this could be:

```
return "span" if @heading_level == 0

[*1..6].include?(@heading_level) ? "h#{@heading_level}" : "h2"
```

That's probably clearer but I figured most of the time the code will be returning a heading, not a span, so I put it this way around.

Reply:
Thanks for the explanation @andysellick